src/main/java/org/xstefank/verification/VerificationHandler.java

```diff
@@ -0,0 +1,32 @@
+package org.xstefank.verification;
+
+import org.xstefank.model.yaml.FormatConfig;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VerificationHandler {
+
+    private List<Verification> verifications;
+
+    public VerificationHandler() {
+        verifications = registerVerifications();
+    }
+
+    public void verifyConfiguration(FormatConfig formatConfig) throws BadFormatException {
+        for (Verification verification : verifications) {
+            if (verification.verify(formatConfig))
+                continue;
+            else
+                throw new BadFormatException(verification.getErrorMessage());
+        }
+    }
+
+    private List<Verification> registerVerifications() {
+        List<Verification> verifications = new ArrayList<>();
+
+        verifications.add(new RepositoryFormatVerification());
```

Comment:
can you come up with one or two more verifications so all this effort is justified please?